### ISSUE TITLE
Penthouse error blocks generating file

### ISSUE THREAD

**Issue by schoenwaldnils** [addyosmani/critical#172]
Hi, I get following error where I sadly have no idea where the problem is:
```
Unhandled rejection TypeError: Bad argument
    at TypeError (native)
    at ChildProcess.spawn (internal/child_process.js:289:26)
    at exports.spawn (child_process.js:380:9)
    at generateCriticalCss (/Users/nils/repos/schoenwaldnils/blog/node_modules/penthouse/lib/index.js:95:10)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

I use jekyll and generate the site to `_site` where the `index.html` lies.
I try to generate the `main-critical.css`.

My gulp task:
```
gulp.task('critical', function (cb) {
  critical.generate({
    inline: false,
    base: '_site',
    src: 'index.html',
    dest: 'build/main-critical.css',
    width: 320,
    height: 480,
    minify: true,
  });
});
```

**Comment by addyosmani** [comment]
@schoenwaldnils Are you still running into this error? If so, is your repo for this project up somewhere?

**Comment by schoenwaldnils** [comment]
@addyosmani Yes, I do. But I now saw that ther is a `.steam` missing in the require for gulp.
Also missing in your tutorial:
https://github.com/addyosmani/critical-path-css-demo#tutorial

But even if I add the steam, then I get following:
```
TypeError: critical.generate is not a function
    at Gulp.<anonymous> (/Users/nils/repos/schoenwaldnils/blog/gulpfile.babel.js:298:12)
    at module.exports (/Users/nils/repos/schoenwaldnils/blog/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/nils/repos/schoenwaldnils/blog/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/nils/repos/schoenwaldnils/blog/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/Users/nils/repos/schoenwaldnils/blog/node_modules/orchestrator/index.js:134:8)
    at /usr/local/lib/node_modules/gulp/bin/gulp.js:129:20
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:592:11)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

**Comment by bezoerb** [comment]
@schoenwaldnils the `stream` function wraps `generate` for gulp. You can use it like this:
```
const log = require('gulp-util').log;
const colors = require('gulp-util').colors;
const critical = require('critical').stream;
const rename = require("gulp-rename");

// Generate & Inline Critical-path CSS
gulp.task('critical', function () {
    return gulp.src('_site/index.html')
        .pipe(critical({
              base: '_site'
              width: 320,
              height: 480,
              minify: true,
        })).on('error', function(err) { log(err.message); })
        .pipe(rename('main-critical.css'))
        .pipe(gulp.dest('build'));
});
```

To get a little more verbose you can toggle the debug mode on the CLI with
```bash
export DEBUG="critical:*"
gulp critical
```

Maybe the debug output leads us to the cause of your issue

**Comment by jonathanmelville** [comment]
Sadly I'm having the same problem and I'm not able to get past it:

```
[09:27:06] Starting 'critical'...
[09:27:07] Finished 'critical' after 1.58 s
Unhandled rejection TypeError: Bad argument
    at ChildProcess.spawn (internal/child_process.js:289:26)
    at exports.spawn (child_process.js:385:9)
    at generateCriticalCss (/Users/jonathanmelville/boxes/atlantaballet/node_modules/penthouse/lib/index.js:95:10)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**Comment by scottsweb** [comment]
Seeing similar errors on my build today too:

```
Unhandled rejection TypeError: Bad argument
    at TypeError (native)
    at ChildProcess.spawn (internal/child_process.js:274:26)
    at exports.spawn (child_process.js:362:9)
    at generateCriticalCss (/home/scott/Sites/vip-go-admin-console/node_modules/penthouse/lib/index.js:102:10)
    at process._tickCallback (node.js:368:9)
```

**Comment by danimalweb** [comment]
I am getting the same error in my build.

```
TypeError: Bad argument
    at TypeError (native)
    at ChildProcess.spawn (internal/child_process.js:278:26)
    at exports.spawn (child_process.js:367:9)
    at generateCriticalCss (/Users/daniel/Private/admin/santabarbaradesigns.dev/public_html/wp-content/themes/santa-barbara-designs/node_modules/penthouse/lib/index.js:102:10)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**Comment by YewTreeWeb** [comment]
I am also getting the error for when I try with Shopify and Jekyll

**Comment by pocketjoso** [comment]
@YewTreeWeb Something must have gone wrong in your install step. The missing config file is [here](https://github.com/pocketjoso/penthouse/blob/master/src/phantomjs/config.json).

**Comment by jadchaar** [comment]
Had this error as well, and it seems like an install issue like pocketjoso said. I suspect it was an issue with installing the package and getting a halt with a "run as root/administrator" prompt. This could have corrupted the installation. I fixed it by clean installing the package with the following commands:
`sudo npm uninstall critical`
`sudo npm install critical --save-dev`

**Comment by pocketjoso** [comment]
Regarding the install error, I just fixed a (new) bug with the missing `config.json` file in [0.11.11](https://github.com/pocketjoso/penthouse/releases/tag/v0.11.11) - so if you re-install criticial/penthouse you should get the fix for this problem.

**Comment by YewTreeWeb** [comment]
@pocketjoso Thank you for the sending me the link to the config and for the bug fix.

**Comment by bezoerb** [comment]
Closing as this one seems to be fixed by [`penthouse v0.11.11`](https://github.com/pocketjoso/penthouse/releases/tag/v0.11.11)

**Comment by bezoerb** [comment]
@pocketjoso Thanks for your work!
